src/doc/reference.md

```diff
@@ -3769,7 +3769,20 @@ impl Printable for String {
 
 The notation `&self` is a shorthand for `self: &Self`. In this case,
 in the impl, `Self` refers to the value of type `String` that is the
-receiver for a call to the method `make_string`.
+receiver for a call to the method `make_string`. Thus, it could also
+have been written:
+
+```
+# trait Printable {
+#     fn make_string(&self) -> String;
+# }
+# 
+impl Printable for String {
+    fn make_string(&self) -> Self {
```

Comment:
Why returning `Self` whereas the trait method return `String`? To be more clear: even if you implement the trait on `String`, the method should still returns `String` and not `Self` for more clarity.